addons/l10n_pk_edi/models/product_product.py

```diff
@@ -0,0 +1,30 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    l10n_pk_edi_sale_type = fields.Many2one('l10n.pk.product.type', 'Sale Type',
+        compute='_compute_l10n_pk_edi_sale_type', required=True, readonly=False, store=True,
+    )
+    l10n_pk_edi_schedule_code = fields.Many2one('l10n.pk.schedule.code', 'Schedule Code',
+        compute='_compute_l10n_pk_edi_schedule_code', required=True, readonly=False, store=True,
+    )
+
+    @api.depends('type')
+    def _compute_l10n_pk_edi_sale_type(self):
+        for line in self:
+            # Set 'code' to 'T1000017' if line.type is 'consu'. 'T1000018' if line.type is 'service'
+            # Otherwise, default to 'T1000083' which represent 'Other'
+            code = 'T1000017' if line.type == 'consu' else 'T1000018' if line.type == 'service' else 'T1000083'
+            line.l10n_pk_edi_sale_type = self.env['l10n.pk.product.type'].search(
+                [('sale_type', '=', code)]
+            ).id
```

Comment:
```suggestion
        def _get_pk_sale_type(type):
            match type:
                case 'consu':
                    return 'T1000017'
                case 'service':
                    return 'T1000018'
                case _:
                    # Otherwise, default to 'T1000083' which represent 'Other'
                    return 'T1000083'
        for line in self:
            line.l10n_pk_edi_sale_type = self.env['l10n.pk.product.type'].search(
                [('sale_type', '=', _get_pk_sale_type(line.type))],
                limit=1
            ).id
```